keps/sig-storage/361-local-ephemeral-storage-isolation/README.md

```diff
@@ -0,0 +1,460 @@
+  For example, check `spec.containers[].resources.limits.ephemeral-storage` of each container.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+
+- [x] Metrics
+  - storage metrics from summary API
+    Node-level: NodeStats {Fs (FsStats) }
+    Pod-level: PodStats.VolumeStats {Name, PVCRef, FsStats }
+    Container-level: ContainerStats {Rootfs (FsStats), Logs (FsStats) } 
+  - Components exposing the metric: kubelet
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+
+  - 99.9% of volume stats calculation will cost less than 1s or even 500ms.
+  It can be calculated by `kubelet_volume_metric_collection_duration_seconds` metrics.
+
+* **Are there any missing metrics that would be useful to have to improve observability of this feature? **
+
+  - Yes, there are no histogram metrics for each volume. The above metric was grouped by volume types because
+    the cost for every volume is too expensive.
+
+### Dependencies
+* **Does this feature depend on any specific services running in the cluster? **
+
+  - No
+
+
+### Scalability
+* **Will enabling / using this feature result in any new API calls?**
+  - No.
+
+* **Will enabling / using this feature result in introducing new API types?**
+  - new resource type API "ResourceEphemeralStorage"
+
+* **Will enabling / using this feature result in any new calls to the cloud
+provider?**
+  - No.
+
+* **Will enabling / using this feature result in increasing size or count of
+the existing API objects?**
+  - No.
+
+* **Will enabling / using this feature result in increasing time taken by any
+operations covered by [existing SLIs/SLOs]?**
+  - No.
+
+* **Will enabling / using this feature result in non-negligible increase of
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+  - Yes. It will use CPU time and IO during ephemeral storage monitoring.
+
+### Troubleshooting
+
+<!--
+This section must be completed when targeting beta to a release.
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+- kubelet can continue to monitor disk usage, but pod eviction will be impacted.
+
+###### What are other known failure modes?
+
+1. If the ephemeral storage limitation is reached, the pod will be evicted by kubelet.
+
+2. It should skip when the image is not configured correctly (unsupported FS or quota not enabled).
+
+3. For "out of space" failure, kublet eviction should be triggered.
+
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+- Restart kubelet and wait for 1 minute to make the SLOs clear.(The volume stats checking interval is determined by kubelet flag `volumeStatsAggPeriod`(default 1m).)
```

Comment:
How does this help determine the problem? The intent behind this question is to help with troubleshooting. What should an operator do to determine the underlying issue, if the SLO is not met?